app/api/services/core/patch_params.py

```diff
@@ -0,0 +1,12 @@
+import dataclasses
+from typing import Generic, TypeVar
+
+ResourceType = TypeVar("ResourceType")
```

Comment:
Is resource the right word? I didn't think of the request params on reading this at first glance, I'd say adding a `bind=` to it would help, but there's no dataclass class to bind to unfortunately.